### PR TITLE
Kustomize for Kubernetes deployment

### PR DESCRIPTION
**Pull request by marccampbell**

### CODE REVIEW

File: kustomize/overlays/production/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: extensions/v1beta1
```
Comment (emosbaugh) [inline kustomize/overlays/production/deployment.yaml]: why deployment overlay when they are the same?

File: kustomize/base/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+spec:
+  template:
+    metadata:
+      labels:
+        app: install-scripts
+    spec:
+      containers:
+        - name: install-scripts
+          image: replicated/install-scripts
+          imagePullPolicy: IfNotPresent
```
Comment (emosbaugh) [inline kustomize/base/deployment.yaml]: why not just add env here and overlay dev?

File: .circleci/config.yml [hunk 3]
```diff
@@ -136,7 +136,48 @@ jobs:
...
             fi
 
-  deploy_production:
+  deploy_staging_eks:
+    <<: *defaults_py
+    steps:
+      - setup_remote_docker
+
+      - attach_workspace:
+          at: .
```
Comment (emosbaugh) [inline .circleci/config.yml]: why are you attaching the workspace?

File: kustomize/base/deployment.yaml (hunk 2 above)
Comment (marccampbell) [reply kustomize/base/deployment.yaml]: It's a design decision, what's the right answer?  I'm torn on this, what should the default behavior be when there's no overlay set?  dev i think.

File: .circleci/config.yml (hunk 3 above)
Comment (marccampbell) [reply .circleci/config.yml]: thanks.  to get the /kustomize directory, but i missed persisting that.  added.

**Pull request merged**